README.md

```diff
@@ -1,64 +1,129 @@
+
+2. **Relayer Service**:
+   - Implemented in Golang to automate root propagation.
+   - Runs on a cron job schedule, ensuring regular updates.
+   - Tested and ready for deployment in both development and production environments.
+
+3. **Supported Networks**:
+   - Fully operational on Optimism, Base, and Polygon PoS.
+   - Initial support for Scroll, with plans for expanding to more EVM-compatible networks.
+
+4. **Future Proofing**:
+   - The project is designed to integrate with upcoming storage proof solutions like Axiom and Herodotus.
+   - Plans to expand support to a broader range of networks as these technologies mature.
+
+5. **Documentation and Developer Support**:
+   - Comprehensive documentation available for setup, deployment, and usage.
+   - Active community and GitHub repository for contributions and support.
+
+---
+
+## Detailed Documentation
+
+### 1. Architecture Overview
+
+The **Worldcoin Scroll Bridge** architecture is designed to ensure secure and efficient cross-chain communication. The primary components include:
+
+- **Mainnet Contract**: The contract on Ethereum mainnet fetches the latest World ID Merkle tree root from the `WorldIDIdentityManagerImplV1` contract using the `latestRoot()` method. This root is then sent to the target L2 networks using the native L1 <> L2 messaging layer.
+  
+- **L2 Contract**: The contract on the L2 network receives the propagated root and updates its state. Ownership and authorization checks are handled by the `CrossDomainOwnable3` contract, ensuring only the mainnet contract can initiate root propagation.
+
+- **Relayer Service**: A Golang-based service that periodically calls the `propagateRoot()` function on the L2 contract. It is responsible for ensuring that the World ID root is consistently propagated to the target networks.
+
+### 2. Supported Networks
+
+The **Worldcoin Scroll Bridge** currently supports the following networks:
+
+- **Polygon PoS**: Provides backward compatibility with earlier versions of the World ID system.
+- **Optimism**: Uses the Optimism cross-domain messenger for root propagation.
+- **Base**: Fully integrated with the World ID system.
+- **Scroll**: Initial support with ongoing improvements and optimizations.
+
+### 3. Future Integrations
+
+The bridge is designed to support additional networks in the future, with a focus on storage proof solutions such as Axiom and Herodotus. These integrations will enable seamless and cost-effective cross-chain identity verification, expanding World ID's reach to a wider range of blockchain environments.
+
+### 4. Relayer Service Implementation
+
+The **Relayer Service** automates the process of propagating the World ID root from Ethereum mainnet to the Scroll network. Key implementation details include:
+
+- **Golang Environment Setup**: The service is implemented in Golang, ensuring high performance and reliability. It uses a cron job to schedule periodic calls to the `propagateRoot()` function.
```

Comment:
Our relayer service is not implemented in golang, but rust: https://github.com/dragan2234/worldcoin-scroll-bridge/pull/17

@Gift-Naomi